### PR TITLE
prevent use of protected metadata fields

### PR DESCRIPTION
**Pull request by mixmix**
Does this one simple thing.

Supercedes: https://github.com/ssbc/ssb-meta-feeds/pull/77

### CODE REVIEW

File: lookup.js [hunk 1]
```diff
@@ -88,16 +89,7 @@ exports.init = function (sbot, config) {
...
-      'metafeed',
-      'feedpurpose',
-      'type',
-      'tangles',
-      'reason',
-      'subfeed',
-      'nonce',
-    ]
-    const keys = Object.keys(content).filter((k) => !NOT_METADATA.includes(k))
+    const keys = Object.keys(content).filter((k) => !NOT_METADATA.has(k))
```
Comment (arj03) [inline lookup.js:92]: why not includes here? It seems to me that has in not an options on array (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) and thus operates on object that just happens to work? Or am I missing something?

File: test/api/find-or-create.test.js [hunk 2]
```diff
@@ -43,27 +43,3 @@ test('findOrCreate', (t) => {
     })
   })
 })
-
-test('findOrCreate (metadata.recps)', (t) => {
```
Comment (arj03) [inline test/api/find-or-create.test.js:47]: just to be sure I'm reading this right, we don't have any tests for encrypted metafeeds after this commit? AH yes, this is now https://github.com/ssbc/ssb-meta-feeds/pull/88. Okay makes sense 👍

File: constants.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+// SPDX-FileCopyrightText: 2022 Mix Irving
+//
+// SPDX-License-Identifier: LGPL-3.0-only
+
+const BB1 = 'bendybutt-v1'
+const v1Details = { feedpurpose: 'v1', feedformat: BB1 }
+const NOT_METADATA = new Set([
```
Comment (mixmix) [inline constants.js:7]: Set

File: lookup.js (hunk 1 above)
Comment (arj03) [reply lookup.js:92]: Ah, did not see that. Perfect

**Review by arj03 — APPROVED** [review]
Nice refactor, thanks

**Pull request merged**